### DISCUSSION TITLE
How to add a plugin to Plugins Manual

### DISCUSSION THREAD

**Discussion by rgleason** [opencpn-manuals/main#20] [Q&A] [answered]
1. sources folder is empty
2.  cmd  antora site.yml  fails.
3.  to update the sources-state list's SHA  did  ./sources.state.sh update
4. to populate the sources folder with those 'manua; onlyl" repos listed iin sources-state  did ./sources-state.sh restore
5. because sources is empty it takes time downloading.
6. however errors are showing

**For example**
```
Restoring into weatherfax_pi
Cloning into 'weatherfax_pi'...
remote: Enumerating objects: 66, done.
remote: Counting objects: 100% (66/66), done.
remote: Compressing objects: 100% (51/51), done.
remote: Total 66 (delta 0), reused 55 (delta 0), pack-reused 0 eceiving objects:  53% (35/66)
Receiving objects: 100% (66/66), 17.15 KiB | 2.14 MiB/s, done.
...
Resolving deltas: 100% (15/15), done.
Updating files: 100% (78/78), done.
fatal: invalid refspec '86ef95e9ef2bf52675b1c0f091b984500315d3dc?'
error: pathspec 'FETCH_HEAD' did not match any file(s) known to git
```

The last two lines  fatal and error?

The entire process ended this way

```
Receiving objects: 100% (5/5), 15.95 KiB | 7.97 MiB/s, done.
remote: Enumerating objects: 30, done.
remote: Counting objects: 100% (30/30), done.
remote: Compressing objects: 100% (25/25), done.
Receiving objects: 100% (30/30), 151.20 KiB | 392.00 KiB/s, done.

Resolving deltas: 100% (6/6), done.
Updating files: 100% (35/35), done.
fatal: invalid refspec '44e8b986e423d7033d83b40a9ff7b0f40f205768?'
error: pathspec 'FETCH_HEAD' did not match any file(s) known to git

Rick@Dart MINGW64 ~/documents/github/OM-plugins (main)
```

Should I believe I've got a complete set in "source" directory?
[sources-state-restore-results.txt](https://github.com/opencpn-manuals/main/files/7196552/sources-state-restore-results.txt)

**Comment by rgleason** [comment]
CMD  antora site.yml  appears to complete with no bellyaching.
So those "Fatal" and "Errors" are false negatives?

Question not an answer!

**Comment by rgleason** [comment]
Also loading the docs  index.html  and looking it seems complete.

Should we just run  "./sources--state.sh restore"   ONLY once despite all the FATAL and ERRORS?

This is a question, not an ansswer!

**Comment by leamas** [comment]
> fatal: invalid refspec '44e8b986e423d7033d83b40a9ff7b0f40f205768?'
> Should I believe I've got a complete set in "source" directory?

No, but I assume you know that now. Otherwise, if youy don't get around in some other way, remove all directories under sources/ (not .gitkeep) and re-run

**Comment by leamas** [comment]
> So those "Fatal" and "Errors" are false negatives?

Which?

**Reply by rgleason** [reply]
Answer: The ones above.   

The sources folder is empty  (I can also remove it myself if needed due to errors)

1. First run  **./source.state.sh update**     
        - Can I run this twice?  
        - How do I know if it finished correctly?
        - What does it update?
        - Does the  _sources.state_ file use the  SHA from each plugins most recent commit? 
        - **ANSWER**: No. First run ./source.state.sh restore
2. **sources-state**   file listing all the plugins exact name,  path and commit
        - Can/should we change/remove the contents of this file?
            - **ANSWER:** It is possible, but then you need to know what you are doing. The safe route is the one I described: put what you want to use in sources/ and use ./source-state.sh save to update sources.state 
        - Under what circumstances should it be changed?
        - Is this the file that is updated by the command _./sources-state.sh update_?
3. **sources** folder contains the immediate local versions of manuals of all the plugins for faster processing  **RESPONSE:** It's not about speed, it's about using them without having them included in the git repo (which would make it big and clumsy). In any case, without sources, no build.
        - Does _./source.state.sh update_  also update these local manual files too?
        - Can we delete all these folders if there are errors?
        - What do we do when we have deleted these manual folders?
        - What does _source.state.sh restore_ do?
4. **site.yml** contains the parent structure for the plugin manual
        - To add a new plugin manual we need to add a record similar to the other plugins?
        - If we have empty  _sources_   folder and empty  _sources.state_ file what is the best way to generate the necessary entries?
        - How is the _sources.state_ file generated from _site.yml_?
        - How is _sources_ directory generated from _sources.state_ file?
 5. **/.sources.state.sh** script operates on what files? 
         - _./sources.state.sh update_  uses _site.yml_ to create the  _sources.state_ file only?
         -_./sources.state.sh restore_   uses _sources.state_ file to download distributed PI manuals into  the _sources_ folder?
        - What does _./sources.state.sh save_ do?  What does it save to where?
        - Can the _./sources.state.sh_  <options>  be used several times without causing problems?
        - What else does _./sources.state.sh_ do?
        - **ANSWER**   **sources.state** and the directories under **sources**:
 
           1.  **restore** copies state from _sources.state_ to _sources_
            2. **update** updates one or all directories in _sources/_          
            3. **save** copies the state in _sources_ to _sources.state_
        
6.  Which of these ./source.state.sh options would cause my local site.yml  and source.state.sh to disappear as they did last night?

**Comment by leamas** [comment]
> First run ./source.state.sh update

No. First run  `./source.state.sh restore`

>  Can/should we change/remove the contents of this file?

It is possible, but then you need to know what you are doing. The safe route is the one I described: put  what you want to use in sources/ and use `./source-state.sh save` to update sources.state

> sources folder contains the immediate local versions of manuals of all the plugins for faster processing

It's not about speed, it's about using them without having them included in the git repo (which would make it big and clumsy). In any case, without sources, no build.

> .sources.state.sh script operates on what files?

sources.state and the directories under sources:
    - restore copies state from sources.state to sources
    - update updates one or all directories in sources/
    - save copies the state in sources to sources.state

**Comment by rgleason** [comment]
> First run _./source.state.sh restore_   sources.state --> sources

So that synchronizes _sources_ folders to all the distributed plugin manuals.
When what I want is Dashboard_tactics in there in sources.
Do I just copy manuals from my repos into sources folder?
Then do _./sources-state.sh save_  which copies the state in _sources_  folder to _sources.state_
So site.yml is not used at all for _sources-state file_ or _sources folder._?

>  _save_ copies the state in _sources_  folder to _sources.state_

This is totally the reverse of what I thought! How does it find the SHA then? 

Is the correct process to add a new plugin manual?
:
1. Modify site.yml  to add plugin
2. Modify _..\manual\master-plugins\modules\ROOT\pages\plugins.adoc_  to add plugin link
3. Modify _..\manual\master-plugins\modules\ROOT\partials\nav-plugins.adoc  to add plugin link
4. Run  _./source.state.sh restore_   To copy state from sources.state --> sources
5. Copy github.com<user>/<plugin>/manual  to opencpn-manuals/plugins/sources/ ?
6. Run  antora site.yml and then check the html manual for the plugins
7. Run  _./sources-state.sh save_  which copies the state in _sources_  folder back to _sources.state_ 
8. Run  _./sources-state.sh update_  which updates one or all directories in _sources/_  for a final check.
9. Run _check-sources_
10. Run antora site.yml a final time  checking your local html
11. Then commit
12. $ git status
13. $ git add site.yml
14. $ git add sources.state
15. $ git add manual/
16. $ git commit -m "Update sources to latest version"  or whatever
17. $ git push -f origin main:build
18. Then look at github actions completion and the website html   push also to main
19. $ git remote update origin
20. $ git rebase origin/main
21. $ git commit -m "[full-linkcheck] empty commit" --allow-empty
22. Response like this  [main 7f1c991] [full-linkcheck] empty commit
23. $ git push -f origin main:build works
24. Done

**Comment by rgleason** [comment, marked as answer]
Note this system has changed all plugins now moved to opencpn-manuals/main  and sources.state system has been removed.
 git push -f origin HEAD:build does the job of gathering all the latest edits of the manuals. If you look at site.yml for opencpn-manuals/main you will see the github sources listed.



### What files does _./source-state.sh  (argument)_  script operate on?

**ANSWER** _sources.state_ file and the directories under _sources_ folder:

- Script: _./source-state.sh  (option)_ 
- **restore** copies state from _sources.state_ file --> _sources_ folders
- **update** updates one or all directories in _sources/_ folders with current commit in the distributed plugins "manual".         
- **save** copies the state in _sources_ folders --> _sources.state_ file

### How to add a plugin to the Plugins Manual

#### First clone the opencpn-manual/plugins repo to your local git folder

- git clone git@github.com/opencpn-manual/plugins.git    (optional local folder name)

#### Add the plugin link to navigation folders

- Modify site.yml to add plugin
- Modify _../manual/master-plugins\modules/ROOT/pages/plugins.adoc_ to add plugin link
- Modify _../manual/master-plugins\modules/ROOT/partials/nav-plugins.adoc_ to add plugin link
- Modify _../manual/master-plugins/modules/ROOT/partials/nav-all-plugins.adoc_  to add plugin link (Keep Sorted!)

#### Restore _sources_ folder (PI manuals) from _sources-state_ file

- Run _./source-state.sh restore_  or _.source-state.sh restore <plugin_name_pi>_ To copy state from _sources.state_ --> _sources_
- If Fatal/Errors result, delete _sources_ recently changed plugins (leave .gitkeep) and try rerunning restore.
- If there are still "Fatals" or "Invalid refspecs",  Remove all directories under sources/ (not .gitkeep) and re-run restore
- Note deleting all plugin folders in sources is about 200mb so best to have fast internet
- If Fatal/Errors still result, the sources-state file is bad,
- Two options,  Check out it from a working commit, or depending on what you want...
- Run _./source-state.sh update_  to align _sources_ folders to be current with distributed github.com plugin commits
- If this completes without error, then bring the sources-state file current with
- Run  ./source-state.sh save
- Then run _./source-state.sh restore_ and it should complete nicely with no errors.
-  You then may want to commit the new sources.state file to build and then main branches.
- Here is an example of good output after some errors  [Success-dashboard_tactics.txt](https://github.com/opencpn-manuals/main/files/7211884/Success-dashboard_tactics.txt)

####  Clone your entire plugin to _**sources**_

- You must clone your repo from github, it's the only way to know you refer to the right commit.
- Navigate locally to the _/opencpn-manuals/plugins/sources/_ directory and then clone the new plugin
- It doesn't matter if its the complete plugin
- git clone git@github.com/(user)/(plugin)

#### Run antora and check html locally.

- Run antora site.yml 
- Then check the html under docs index.html  find the manual for the plugin

####   Save _sources_ folder state to back to _sources.state_ file and final check.

- How does the dashboard_manual get included?  -  _./source-state.sh_ save saves everything under sources into sources.state. 
- So, if you have cloned a repository into sources folder, it will be included by save.
- Run _./source-state.sh save_ which copies the state in sources folder back to sources.state
- Run _check-sources_ [check-sources-output.txt](https://github.com/opencpn-manuals/main/files/7212174/check-sources-output.txt)
- Just make sure your plugin checks out, don't worry about others, this commit is for your PI only.
- Run _antora site.yml_ a final time to check your local html  results.
 
#### Then commit main to _**build**_ branch

- $ git status
- $ git add site.yml
- $ git add sources.state
- $ git add manual/
- $ git commit -m "sources, site.yml: Add new( plugin_name)" or clear commit title as appropriate
- $ git push -f origin main:build
- Then look at the https://github.com/opencpn-manuals/plugins _build_ branch last commits under github _Actions_  for errors.
- Then look carefully at the antora output at https://opencpn-manuals.github.io/plugins/opencpn-plugins/0.1/index.html

#### Then [Full-Linkcheck] with an empty commit

- $ git commit -m "[full-linkcheck] empty commit" --allow-empty
- Response: like this [main 7f1c991] [full-linkcheck] empty commit
- $ git push -f origin main:build 
- Then look at the https://github.com/opencpn-manuals/plugins _build_ branch last commits under github _Actions_  for errors.
- Then look carefully at the antora output at https://opencpn-manuals.github.io/plugins/opencpn-plugins/0.1/index.html

#### Then commit to _**main**_ branch

- $ git remote update origin
- $ git rebase origin/main
- $ git push  origin main:main     (Don't use the -f option!)
-  Pushing to main does not generate any HTML so nothing to check there, or in Actions log.

Also it would be helpful to read  [Understanding the github  Plugins Manual Repository](https://opencpn-manuals.github.io/plugins/opencpn-plugins/0.1/advanced/advanced.html)

#### Other useful git commands

-  git log --oneline -10
-  git rev-parse --short HEAD

#### Forget your local changes, get back to main branch HEAD

- git checkout main
- git remote update origin
- git reset --hard origin/main

OR

- git fetch
- git reset --hard HEAD
- git merge origin/$CURRENT_BRANCH

OR

- git fetch
- git reset --hard HEAD
- git merge '@{u}' 

 
 #### NOTE Always include your commit SHA if you are reporting a problem under Issues!

**Reply by leamas** [reply]
> Copy github.com//manual to opencpn-manuals/plugins/sources/ ?

No. Checkout the complete repo: `cd sources; git clone ...`

**Reply by rgleason** [reply]
Well yes. that would be first.  I will add that..

**Reply by rgleason** [reply]
Copy github.com//manual to opencpn-manuals/plugins/sources/ ?

I don't understand what you are sayiing for this step. Can you elaborate?  See above whiich is revised.

**Reply by rgleason** [reply]
Found this response from Alec

https://github.com/opencpn-manuals/main/discussions/16#discussioncomment-1362355

which helps, but it is still not clear.  How does the dashboard_manual get included?

**Reply by leamas** [reply]
./source-state.sh save saves everything under sources into sources.state. So, if you have cloned a repository there it will be included by _save_.

**Reply by leamas** [reply]
> Run ./sources-state.sh update which updates one or all directories in sources/ for a final check.
> Run ./sources-state.sh save which copies the state in sources folder back to sources.state

Please drop this. It will mess up the commit with the update of other plugins. Makes no sense.

**Reply by leamas** [reply]
> Manually copy your github.com/(plugin)/manual to opencpn-manuals/plugins/sources/

No. You should clone your repo from github, it's the only way to know you refer to the right commit

SEE:  https://github.com/opencpn-manuals/main/discussions/20#discussioncomment-1363934

**Reply by leamas** [reply]
> Run check-sources

Perhaps. But only care about the output for  the plugin you are about to add, the rest is another issue and belongs in another commit. The trick is to create a focused commit which adds the new plugin

**Reply by leamas** [reply]
>  git commit -m "Update sources to latest version" or whatever

Definitely not. The message should say something like "sources, site.yml: Add new plugin foo". Please give the commit messages some more thought...

**Reply by leamas** [reply]
> git add site.yml
$ git add sources.state
$ git add manual/

or

    $ git add sources.state site.yml manual

**Reply by rgleason** [reply]
>     Manually copy your github.com/(plugin)/manual to opencpn-manuals/plugins/sources/
>
> No. You should clone your repo from github, it's the only way to know you refer to the right commit

Changed

>     Run ./sources-state.sh update which updates one or all directories in sources/ for a final check.
>     Run ./sources-state.sh save which copies the state in sources folder back to sources.state
> 
> Please drop this. It will mess up the commit with the update of other plugins. Makes no sense.

Removed:  I think thats what happened when I lost everything locally.  site.yml  sources-state.sh etc disappeared.
I will now clone opencpn-manuals/plugins again  (fresh)  and try this if you think the above looks ok

https://github.com/opencpn-manuals/main/discussions/20#discussioncomment-1363128

**Reply by leamas** [reply]
You also need to add the new plugin to the brand new _manual/master-plugins/modules/ROOT/partials/nav-all-plugins.adoc_ navigation page. Please keep that page sorted...

**Reply by leamas** [reply]
> Then commit main to main branch
    $ git remote update origin
    $ git rebase origin/main
    $ git push origin main:build (Don't use the -f option!)
    Then look at github actions completion and the website html push also to main

Makes no sense. Should be:

    $ git remote update origin
    $ git rebase origin/main
    $ git push origin main:main (Don't use the -f option!)

Pushing to main does *not* generate any HTML so nothing to check there, or in Actions log.

**Reply by leamas** [reply]
> Run antora site.yml a final time checking your local html

Basically a duplicate and does not really add any value. Better to spend some more time checking the output of the first run of `antora site.yml`

**Reply by rgleason** [reply]
Thanks Alec, I've made those changes.

**Reply by leamas** [reply]
Nope, you have not. Please check again.

**Reply by rgleason** [reply]
Alec I've made those changes, I have checked.  Hit rewrite on your browser?

**Comment by rgleason** [comment]
Alec's Answer:

The "right" way to use sources and source-state.sh is like this (omitting all shortcuts):

-  Checkout a version of _sources.state_ file which is known to work.
-  Remove all directories under sources (but keep .gitkeep)
-  Restore the sources: ./source-state.sh restore

If you want to update for example weather-routing_pi:

     $ ./source-state.sh update weather-routing_pi

Make a local build and check the results: antora site.yml

Record the current state and commit it

    $ ./source-state.sh save
    $ git add sources.state
    $ git commit -m "sources: Update weather-routing_pi."

And that's it... This is an ordered way to get the correct info into sources.state. It takes some time, but works.

**Reply by rgleason** [reply]
This is very Useful information but it does not say how to ADD a Plugin Manual

**Reply by leamas** [reply]
Similar. restore all sources, go to the sources directory and clone the new plugin, tactics_pi in your case. It doesn't matter it's the complete plugin. Then make a test build. If everything is fine, run ./source-state.sh save and commit sources.state.

Done!

**Reply by rgleason** [reply]
I think I've got it. Another use for git.

**Comment by rgleason** [comment]
Alec, I have been following the instructions starting with a fresh git clone.
Restore sources manuals

```
    Run ./source.state.sh restore To copy state from sources.state --> sources
    If that has "Fatals" or "Invalid refspec" Remove all directories under sources/ (not .gitkeep) and re-run restore

```

Alec  Please see attached output..txt   Just about every plugin has "FATAL"
Created with  ./source.state.sh restore > output.txt

[output.txt](https://github.com/opencpn-manuals/main/files/7206446/output.txt)

What next?
Have done this three times, each time having to delete all of sources except  .gitkeep  then rerun  ./sources.state.sh restore

**Reply by leamas** [reply]
Not a single error message in that file. You have probably not redirected stderr.  What commit are you on? Have you pushed it?

**Reply by rgleason** [reply]
Ok, I guess that does not work...  - here is part of what I am seeing in the Bash prompt:
[Fatal-Error-Bash-prompt.txt](https://github.com/opencpn-manuals/main/files/7206693/Fatal-Error-Bash-prompt.txt)
This is very different.  
No I wouldn't push this!   So I don't have SHA
Any ideas?
Thankks

**Reply by rgleason** [reply]
@leamas Do you think it actually worked fine, but I am seeing Fatals and Errors in the Bash prompt that are false negatives?   ..maybe you are asleep by now.

**Reply by rgleason** [reply]
I will try this again with 
Redirect stdout to one file and stderr to another file:
 
  command > out 2>error

**Reply by rgleason** [reply]
[output.txt](https://github.com/opencpn-manuals/main/files/7206865/output.txt)
[error.txt](https://github.com/opencpn-manuals/main/files/7206867/error.txt)

**Reply by rgleason** [reply]
I removed my change to site.yml for dashboard_tactics and done
antora site.yml

Checking all the html of the plugins manuals,  Main and Beta, there are none that are not working!

This is working. The proof is in the pudding here. 
I am going to continue and will push.
The Fatals and Errors are pure doXx IMHO

**Reply by rgleason** [reply]
After cloning dashboard_tactics_pi  antora site.yml  produced the dashboard_tactics_pi manual Ok.

This is the result of check-source
[source-state-sh-save.txt](https://github.com/opencpn-manuals/main/files/7206990/source-state-sh-save.txt)
I am not responsible for the two errors, however dashboard_tactics_pi does pass.
I did add a PR to ocpn_draw to [fix the site.yml start page.](https://github.com/jongough/ocpn_draw_pi/pull/487)
The other error is due to AISradar PR not being accepted.

**Reply by rgleason** [reply]
This pretty much pXXXXX me off.  windvane failed again  ceac8738400d535ab4a383d1d7bf6c395eef5755
Now some other unknown and unfindable error by some other process has created another blockage.
Its happened often enough before so I am pretty sure about that.
ceac8738400d535ab4a383d1d7bf6c395eef5755  build branch
https://github.com/opencpn-manuals/plugins/commit/ceac8738400d535ab4a383d1d7bf6c395eef5755

**Reply by rgleason** [reply]
Rick@Dart MINGW64 ~/documents/github/OM-plugins (main)
$ git remote update origin
Fetching origin

Rick@Dart MINGW64 ~/documents/github/OM-plugins (main)
$ git rebase origin/main
Current branch main is up to date.

Rick@Dart MINGW64 ~/documents/github/OM-plugins (main)
$ git log --oneline -10
ceac873 (HEAD -> main, origin/build) Update sources to add dashboard_tactics
5295200 (origin/main, origin/HEAD) Navigation: Add a "All plugins A-Z" section
fbbaaa1 oesenc [full-linkcheck]
25ea807 w3c-linkcheck.sh: Restore exclude ordering.
59e65ed Fix bad wc3 linkcheck link exclude
affa384 [full-linkcheck] empty commit
ff04b07 [full-linkcheck] empty commit
d47ac45 [full-linkcheck] empty commit
2254bf9 sml edit nav plugins nav-plugins adoc
0bc313f edit workflows table

**Reply by rgleason** [reply]
Error

> From **https://github.com/jongough/windvane_pi**
>  * branch            44e8b986e423d7033d83b40a9ff7b0f40f205768 -> FETCH_HEAD
> + antora site.yml
> error: **playbook file not found at /home/runner/work/plugins/plugins/site.yml** (path: site.yml, cwd: /home/runner/work/plugins/plugins)
> Add the --stacktrace option to see the cause.
> Error: Process completed with exit code 1.

**Both site.yml and antoral.yml exist  in windvane **

**Reply by leamas** [reply]
And when will you learn that "remove" means **remove** a line, not putting C++ comment marks in front of it. I see this all over the place.

This is git, no changes are lost, and all those commented line are, in general, a pain. If a line should be removed, drop it. It still exists in git.

In this particular case, source-state.sh does not parse the // marks and the results when using them are undefined.   Bit it's an overall issue, I have had to remove // marked lines on other places just because of this strange habit you have.

**Reply by leamas** [reply]
This takes too long time. I have added dashboard_tactics to sources, site.yml and the navigation items in 0fd3f8b and 525f54b  Will update the main manual later.

**Reply by leamas** [reply]
That is, https://github.com/opencpn-manuals/plugins/commit/0fd3f8b5f7b and https://github.com/opencpn-manuals/plugins/commit/525f54b

**Reply by rgleason** [reply]
```
And when will you learn that "remove" means remove a line, not putting C++ comment marks in front of it. I see this all over the place.

This is git, no changes are lost, and all those commented line are, in general, a pain. If a line should be removed, drop it. It still exists in git.

In this particular case, source-state.sh does not parse the // marks and the results when using them are undefined. Bit it's an overall issue, I have had to remove // marked lines on other places just because of this strange habit you have.
```

I am  not the one who put those comments in that nav-plugins.adoc  etct!!!!
You should write this to the person who did that.

**Reply by rgleason** [reply]
I have 
git remote update origin
antora site.yml  works
./source-state.sh restore
   - results in good output (attached). (note that site.yml has been fixed for correct edit_url too)


   
[Success-dashboard_tactics.txt](https://github.com/opencpn-manuals/main/files/7211862/Success-dashboard_tactics.txt)

**Reply by rgleason** [reply]
**Question about Process**

### How to add a plugin to the Plugins Manual

1. First clone the opencpn-manual/plugins repo to your local git folder
2. Restore sources manuals
3. Clone your entire plugin to sources
4. Run antora and check html locally.
5.  Strikeout-  Restore sources manuals - Again - To copy your plugin's state  **<-Essential for your cloned plugin?** Strikeout
NOPE Removed it is WRONG. 
Restore  copies state from _sources.state_ file to _sources_ folders!  Your cloned PI will be removed. 
6. Save sources folder to back to sources.state file final check.
7. Then commit main to build branch
8. Then Full-Linkcheck with an empty commit
9. Other useful git commands
10. NOTE Always include your commit SHA if you are reporting a problem under Issues!

**Comment by rgleason** [comment]
Mike made this PR to Dashboard Tactics  for fixing the site.yml on my local repos   -Thank you Mike
Is this the reason I could not push to the Beta branch without errors?
https://github.com/rgleason/dashboard_tactics_pi/pull/6/commits/ac6a2105aebfeaac4c8eb5fc171093d2799b0c9d

remove      edit_url: https://github.com/**_opencpn-manuals_**/dashboard_tactic**_s_**s_pi/edit/master/{path}
add           edit_url: https://github.com/**_rgleason/_**dashboard_tactic**_s_**s_pi/edit/master/{path}

I must have copied the wrong kind of site.yml !!!   perhaps an internal plugin

Which I will accept, but I also notice that there is an extra "S" in tactics  (probably due to my eyesight).
Can this be fixed before going live I wonder?
I will make the change in my repos.   - Done
